src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_pending_resumption_handler.cc

```diff
@@ -267,6 +269,14 @@ void VehicleInfoPendingResumptionHandler::HandleResumptionSubscriptionRequest(
   SDL_LOG_TRACE("app id " << app.app_id());
   auto& ext = dynamic_cast<VehicleInfoAppExtension&>(extension);
 
+  const auto pending = ext.PendingSubscriptions().GetData();
+  for (const auto& ivi : pending) {
+    if (IsSubscribedAppExist(ivi, application_manager_)) {
+      ext.RemovePendingSubscription(ivi);
+      ext.subscribeToVehicleInfo(ivi);
+    }
+  }
+
```

Comment:
@ychernysheva I believe you may move this code block after `subscriptions.empty()` statement as `subscriptions` holds the same data so you can just reuse it if it is not empty

Reply:
@AKalinich-Luxoft In fact `pending` may be differ from `subscriptions` in terms of number pending items, because during iteration already subscribed params will be deleted from `pending` set, so `subscriptions` may became empty after this iteration. That's why this check for emptiness is placed after this loop, because it's important to may possibility to return in time from this function to prevent duplicate subscriptions.
To make this changes clearer I propose to change for loop this way: 146115a